games/wesnoth/Portfile

```diff
@@ -96,31 +55,25 @@ depends_lib-append      port:boost \
                         port:pcre \
                         port:zlib
 
-post-extract {
-    ln -s ${prefix}/include ${build.dir}/Headers
-    ln -s ${prefix}/lib ${build.dir}
-}
-
-patchfiles              xcode.patch
-
-compiler.cxx_standard   2011
+compiler.cxx_standard   2017
 
-pre-build {
-    system -W ${worksrcpath} "${prefix}/bin/scons -j${build.jobs} translations"
-}
+patchfiles-append       patch-wesnoth-apple-notifications.diff
+patchfiles-append       patch-wesnoth-sdl-main.diff
 
-build.dir               ${worksrcpath}/projectfiles/Xcode
+# src/global.hpp:28:2: error: "Compilation with NDEBUG defined isn't supported, Wesnoth depends on asserts."
+configure.cppflags-append -UNDEBUG
 
-xcode.configuration     Release
-xcode.target            {The Battle for Wesnoth}
+pre-destroot {
+    # Reproduce the logic from app 1.0 PG
```

Comment:
Why is the app portgroup needed? Doesn't the build system already produce an app? If not, how does upstream produce the app for the Mac builds it distributes on its web site?

If the app portgroup is needed, why is reproducing its logic in this portfile necessary?

Reply:
> Why is the app portgroup needed? Doesn't the build system already produce an app? If not, how does upstream produce the app for the Mac builds it distributes on its web site?

Wesnoth includes a CMake build and an Xcode build. (Also an scons build.) According to the docs, the Xcode build requires Xcode 8. Using the CMake build along with the app PG lets us target older systems that cannot run Xcode 8.

> If the app portgroup is needed, why is reproducing its logic in this portfile necessary?

Maybe I am mistaken but I thought that `pre-destroot` in a Portfile would override the `pre-destroot` in the app PG. I am open to suggestions on how to move the NIB file into Contents/Resources without interfering with the app PG logic.

Reply:
> Maybe I am mistaken but I thought that `pre-destroot` in a Portfile would override the `pre-destroot` in the app PG.

You're mistaken! :) All `pre-` and `post-` blocks are cumulative and run in the order in which they were declared.

Reply:
> Using the CMake build along with the app PG lets us target older systems that cannot run Xcode 8.

That's great! Did you check that the Info.plist that the app portgroup generated is sufficiently similar to the Info.plist they ship in their binary?

Reply:
> You're mistaken! :) All pre- and post- blocks are cumulative and run in the order in which they were declared.

Good to know :-)

> That's great! Did you check that the Info.plist that the app portgroup generated is sufficiently similar to the Info.plist they ship in their binary?

There are some minor differences, e.g. copyright string and app category. I have just updated the Portfile to enable the Retina setting. I don't think any of the remaining differences affect the run-time.